[{% octicon arrow-left height:32 class:"right left" vertical-align:middle aria-label:hi %}](FAQs.md) [{% octicon home height:32 class:"right left" aria-label:hi %}](index.md) [{% octicon arrow-right height:32 class:"right left" aria-label:hi %}](APP_DATA.md)

# Appendix: List of Tools used

## Basecalling
 * Guppy (available at Oxford Nanopore Community)
 
## Quality Control
 * [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
 * [PycoQC](https://github.com/a-slide/pycoQC)
 * [MinION_QC](https://github.com/roblanf/minion_qc)


## Read filtering & trimming
 * [Porechop](https://github.com/rrwick/Porechop)
 * [Nanofilt](https://github.com/wdecoster/nanofilt)


## Assemblers
 * [Miniasm](https://github.com/lh3/miniasm)
 * [Flye](https://github.com/fenderglass/Flye)
 * [Shasta](https://github.com/chanzuckerberg/shasta)
 
 
## Read Mappers
 * [Minimap2](https://github.com/lh3/minimap2)
 * [BWA](https://github.com/lh3/bwa)
 
 
## Assembly assessment & visualisation
 * [Assemblytics](http://assemblytics.com)
 * [Mummer](http://mummer.sourceforge.net/)
 * [Bandage](https://rrwick.github.io/Bandage/)
 * [IGV](https://software.broadinstitute.org/software/igv/)
 
 
## Error correction
 * [Racon](https://github.com/isovic/racon)
 * [Minipolish](https://github.com/rrwick/Minipolish)
 * [Medaka](https://github.com/nanoporetech/medaka)
 * [Pilon](https://github.com/broadinstitute/pilon)
 
 
## Variant Calling
 * [Sniffles](https://github.com/fritzsedlazeck/Sniffles)
 
 
## Others
 * [Samtools](https://github.com/samtools)




