# Overview 

[{% octicon arrow-left height:32 class:"right left" vertical-align:middle aria-label:hi %}](index.md) [{% octicon home height:32 class:"right left" aria-label:hi %}](index.md) [{% octicon arrow-right height:32 class:"right left" aria-label:hi %}](SU.md)


## Motivation

The motivation for these tutorials was to give people without bioinformatics experience an entry-level introduction to sequence analysis, specifically whole-genome sequencing (WGS) using long-read data analysis with focus on Oxford Nanopore data. 

As an introduction the tutorials won't cover advanced tool use or workflows but are intended to give people a starting point and basic understanding of general sequence analysis workflows. This includes the general steps but also raise awareness for potential problems and pitfalls. 


## Tutorials

The tutorials are organised in the order they would potentially be executed in a common whole-genome sequencing project. However, they don't have to be executed in a specific order. All tutorials stand alone and can be done without the need to do previous tutorials.

Although the focus of these tutorials is on the analysis of long-read Oxford Nanopore data most steps and tools also apply for other long-read technologies such as PacBio and even Illumina short-reads. All steps and/or tools are marked according to the technologies they can be used for: <img src="figures/ONT.png" height="20px"> for Oxford Nanopore specific tolls/steps, <img src="figures/LR.png" height="20px"> for general Long-Read technologies and <img src="figures/SR.png" height="20px"> for steps/tools that can be used for long-read as well as short-read data.


## Appendix

Additional information about the tutorials, e.g. software list, references and tutorial data, as well as background information  about (long-read) sequence analysis can be found in the Appendix section.
